### ISSUE TITLE
Propose to add constant padding mode to the `torch.nn.functional.grid_sample` function

### ISSUE THREAD

**Issue by TianyiFranklinWang** [pytorch/pytorch#111538]
### 🚀 The feature, motivation and pitch

I'm working on a registration project. When I am using `torch.nn.functional.grid_sample` to apply the displacement field to the moving image, I find I am unable to use constant values like 255 to pad it (at least it is not that intuitive) just like the way in `np.pad`. So I propose to add one. If possible I would really want to implement it myself, though I am a completely new contributor to the pytorch project.

### Alternatives

As a workaround, I pre-pad the image with 255 pixels on the border and use the `border` mode.

### Additional context

Here are two samples,
![`zeros` mode](https://github.com/pytorch/pytorch/assets/54883050/04bc5775-f9c4-4a4a-8b14-2995ce0a41dd)
![pre-pad `border` mode](https://github.com/pytorch/pytorch/assets/54883050/562460b1-c241-42a1-9547-eafec12d52eb)

**Comment by mikaylagawarecki** [comment]
It seems reasonable to add an option similar to padding_mode="zeros" to support this. We would need a way to pass the value however.

**Comment by TianyiFranklinWang** [comment]
We might use [`torch.nn.functional.pad`](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html#torch.nn.functional.pad) as a reference. We can have an optional `value` argument to pass the constant. This method also ensures back compatibility.

**Comment by TianyiFranklinWang** [comment]
Hi @colesbury @mikaylagawarecki, I opened PR #134826 some time ago that implemented the feature and wanted to kindly ask if anyone could review it. I'll be happy to address any feedback or make adjustments as needed.
